src/main/java/com/amplitude/Amplitude.java

```diff
@@ -137,7 +139,11 @@ public synchronized void flushEvents() {
       eventsToSend.clear();
       CompletableFuture.supplyAsync(
           () -> {
-            Response response = httpCall.syncHttpCallWithEventsBuffer(eventsInTransit);
+            Response response = null;
+            try {
+              response = httpCall.syncHttpCallWithEventsBuffer(eventsInTransit);
+            } catch (AmplitudeInvalidAPIKeyException e) {
```

Comment:
I see we are suppressing by catching in here, so for user who accidentally input in invalid apiKey, how would they know if the apiKey is invalid?